client/src/components/FlashCard/CreateFlashCard.js

```diff
@@ -0,0 +1,131 @@
+import React from "react"
+import { Button, Stack, TextField } from "@mui/material"
+import axios from 'axios'
+
+const CreateFlashcard = ({ deckId }) => {
+  
+  const [formValue, setFormValue] = React.useState({})
+  var [errors, setErrors] = React.useState({
+    frontImage: false,
+    frontText: false,
+    backImage: false,
+    backText: false,
+  })
+
+  
+  function validateProperty(name, value) {
+
+    
+    if (name === 'frontImage' || name === 'backImage') {
+      value = value.replace(/\s/g, '').trim() 
+      if (value === '' || value === null) {
+        setErrors({ ...errors, [name]: true })
+        return false;
+      }
+      else if (value.length > 250) {
+        setErrors({ ...errors, [name]: true })
+        return false;
+    }
+    else if (name === 'frontText' || name === 'backText') {
+      value = value.trim()
+      if (value === '' || value === null) {
+        setErrors({ ...errors, [name]: true })
+        return false;
+      }
+      else if (value.length > 500) {
+        setErrors({ ...errors, [name]: true })
+        return false;
+      }
+    }
+    setErrors({ ...errors, [name]: false })
+    return true;
+  }}
+
+  const handleChange = (event) => {
+    event.preventDefault()
+    const currentValues = formValue
+    currentValues[event.target.name] = event.target.value
+    if (validateProperty(event.target.name, event.target.value, errors)) {
+      const currentValues = formValue
+      currentValues[event.target.name] = event.target.value
+      setFormValue(currentValues)
+    }
+    setFormValue(currentValues)
+}
+
+function validateForm(field) {
+  for (var i in errors) {
+    if (errors[i]) return false
+  }
+  return true
+}
+
+
+  const handleSubmit = (event) => {
+    console.log("[CreateFlashcard] onSubmit ", event)
+    event.preventDefault()
+
+    
+
+    if (validateForm()) {
+      
+      axios.post(`http://localhost:8000/decks/${deckId}/cards`, formValue)
+      .then(response => {
+        console.log("[CreateFlashcard] onSubmit response ", response)
+      })
+      .then(response => {
+        window.location.reload()
```

Comment:
Updating the browser location is a dangerous operation. Why do you want to force a reload?